### PR TITLE
new base64-bytestring and webauthn-signatures

### PR DESCRIPTION
**Pull request by imalsogreg**

### CODE REVIEW

File: src/Chainweb/Rosetta/Utils.hs
Comment (edmundnoble) [inline src/Chainweb/Rosetta/Utils.hs:722]: Not sure, maybe ask Linda.

File: src/Chainweb/Pact/SPV.hs
Comment (imalsogreg) [inline src/Chainweb/Pact/SPV.hs:198]: @emilypi and @edmundnoble here is a (verbose) strategy for forking the base64 decoding behavior. This code could use lots of scrutiny!

File: cabal.project
Comment (emilypi) [inline cabal.project:162]: this megaparsec bound seems like a merge artifact

File: src/Chainweb/Pact/SPV.hs
Comment (emilypi) [inline src/Chainweb/Pact/SPV.hs:222]: The reasoning behind this change in particular is that the previous base64-bytestring versions allowed "non-canonical" encodings in the sense that they are not true base64, but will decode to a particular string that will not round trip. Because we can assume that canonicity errors are the result of a structurally sound base64 input, we defer to the lenient decoder, which doesn't enforce canonicity. This is *terribly inefficient*, but since it only runs once or twice on a string of max size 1kb, we're not going to suffer too much.

File: src/Chainweb/Pact/SPV.hs
Comment (emilypi) [inline src/Chainweb/Pact/SPV.hs:250]: These calculations patch the difference between the old decode messages generated from improper padding (e.g. `X===`) and the new ones which fail immediately.

File: src/Chainweb/Pact/SPV.hs
Comment (emilypi) [inline src/Chainweb/Pact/SPV.hs:271]: nice 👍

File: src/Chainweb/Rosetta/Utils.hs
Comment (emilypi) [reply src/Chainweb/Rosetta/Utils.hs:722]: @LindaOrtega

File: chainweb.cabal
Comment (emilypi) [inline chainweb.cabal:520]: does `base16-bytestring` need to be bumped as well, or is that unused in Chainweb?

File: src/Chainweb/Utils.hs
Comment (edmundnoble) [inline src/Chainweb/Utils.hs:288]: ^ Cleanup commented lines

File: src/Chainweb/Version/Mainnet.hs
Comment (edmundnoble) [inline src/Chainweb/Version/Mainnet.hs:144]: For this and testnet replace with `AllChains ForkNever` just to avoid accidentally leaving this in

**Review by edmundnoble — APPROVED** [review]
Good to merge after Pact PR is in.

File: src/Chainweb/Version/Development.hs
Comment (imalsogreg) [inline src/Chainweb/Version/Development.hs:69]: @edmundnoble Should this one, too be `ForkNever`, in this PR?

File: src/Chainweb/Rosetta/Utils.hs
Comment (LindaOrtega) [inline src/Chainweb/Rosetta/Utils.hs:722]:
```suggestion
    toRosettaSigType (Just P.WebAuthn) = Nothing 
    -- TODO: Linda Ortega (09/18/2023) -- Returning `Nothing` to discourage using WebAuthn for Rosetta. `sigToScheme` will eventually throw an error.
```

File: src/Chainweb/Version/Development.hs
Comment (edmundnoble) [reply src/Chainweb/Version/Development.hs:69]: No, this is fine.

Testnet and mainnet are always on, so their block height is advancing and we need to make sure their fork heights roughly match up with specific times. Devnet isn't always on, so this just affects how long it takes until a fresh devnet can use forks.

File: chainweb.cabal
Comment (imalsogreg) [reply chainweb.cabal:520]: Had to be bumped to >= 1.0 for compatibility with `webauthn` lib..

**Review by emilypi — COMMENTED** [review]
I think it would be worth a negative test for Pact SPV to cover our bases.

File: flake.nix
Comment (chessai) [inline flake.nix:5]: why?

i would much rather us pin our own nixpkgs. we have our own cache so we don't need to depend on IOG for that.

**Review by chessai — CHANGES_REQUESTED** [review]

**Review by ak3n — APPROVED** [review]

**Review by chessai — APPROVED** [review]

**Review by emilypi — APPROVED** [review]

**Pull request merged**
